cheatsheets/Forgot_Password_Cheat_Sheet.md

```diff
@@ -1,60 +1,137 @@
+- A user should have multiple recovery codes at any given time to ensure that one of them works (most services provide the user with 10 backup codes).
+
+## Methods Implementation
+
+In order to implement the forgot password service, the developer needs to choose one of the proposed [methods](#methods).
+
+### One Time Password
+
+One Time Password (OTP) is the best method in order to implement a secure forgot password service that triggers as a 2FA functionality.
+
+The two most famous methods are Time-OTP ([TOTP](https://tools.ietf.org/html/rfc6238)), or HMAC-OTP ([HOTP](https://tools.ietf.org/html/rfc4226)). The main difference is in the counter, where TOTP focuses on the Unix time, and HOTP has a counter that gets incremented on every user call to generate the OTP.
+
+For a better description of OTP generation, refer to the [MFA CS](Multifactor_Authentication_Cheat_Sheet.md#something-you-have), where the pros and cons of every implementation are provided.
+
+One implementation can be found over for [Authy](https://www.twilio.com/docs/authy/tutorials/two-factor-authentication-python-flask). If you don't want to rely on applications (such as Authy, Google/Microsoft Authenticator, etc.), there are libraries like pyotp, which helps the developer implement on their own any of the methods discussed in this section.
+
+> OTPs can be sent through other channels as well, such as emails and SMSs. [Various attacks and weaknesses](https://en.wikipedia.org/wiki/SIM_swap_scam) have been identified in SMS that it is [preferrable not to use them for OTPs](https://auth0.com/blog/why-sms-multi-factor-still-matters/).
+
+### URL Tokens or Codes
+
+URL tokens provide access control to the user by sending a URL with a token appended in the querystring, or by sending a code. They are both sent through a side-channel, *e.g.* email.
+
+If a code is provided to the user, the user will have to provide that code manually to the application.
+
+If a URL with a token is provided to the user, the user will have to follow the below steps:
+
+1. Access the URL with the attached token.
+2. Set the token in a cookie (recommended), or the client-storage mechanism that your application uses.
+3. Remove the token from the URL and redirect the user to the password reset service. This ensures that protection against [referer leakage](https://portswigger.net/kb/issues/00500400_cross-domain-referer-leakage).
+4. Let the user create a new password and confirm it. Ensure that the password policy is applied.
+
+### Security Questions
+
+> This method should not be used as the sole method to reset a password, and should be used in conjunction with other methods.
+
+Kindly refer to the [Security Questions Cheat Sheet](Choosing_and_Using_Security_Questions_Cheat_Sheet.md) for further guidance.
```

Comment:
Rephrase to something along the lines of

Security questions should not be used as the sole mechanism for resetting passwords due to PROBLEMS. However, if they are used as a factor then ensure that secure questions are chosen, as discussed in the Security Questions CS.